### ISSUE TITLE
[FEATURE] Please expose a pure player can be custom like give a textureId?

### ISSUE THREAD

**Issue by figbin** [jhomlala/betterplayer#359]
I just want to get the video only, and then I write control widget myself?
because when the player dispose(I need the controller for used few minutes), it make a setState error

```
[VERBOSE-2:ui_dart_state.cc(177)] Unhandled Exception: setState() called after dispose(): _BetterPlayerVideoFitWidgetState#ddbcd(lifecycle state: defunct, not mounted)
This error happens if you call setState() on a State object for a widget that no longer appears in the widget tree (e.g., whose parent widget no longer includes the widget in its build). This error can occur when code calls setState() from a timer or an animation callback.
The preferred solution is to cancel the timer or stop listening to the animation in the dispose() callback. Another solution is to check the "mounted" property of this object before calling setState() to ensure the object is still in the tree.
This error might indicate a memory leak if setState() is being called because another object is retaining a reference to this State object after it has been removed from the tree. To avoid memory leaks, consider breaking the reference to this object during dispose().
#0      State.setState.<anonymous closure> (package:flutter/s<…>
```
officical plugin video_player can't set http headers!

**Comment by figbin** [comment]
It means the Player widget has dispose, but I use this controller setupDataSource.
I will be display new player widget few minutes later!

**Comment by jhomlala** [comment]
@figbin Please follow issue/feature template.

There's nothing wrong with Better Player, you're simply using it wrong way.
If you want to keep BetterPlayerController after your BetterPlayer widget dispose, then set autoDispose flag as false in BetterPlayerConfiguration. This will prevent auto dispose of controller once BetterPlayer disposes.

There won't be feature which exposes clear texture. But you may try to set custom player theme. Here's an example https://github.com/jhomlala/betterplayer/blob/master/example/lib/pages/change_player_theme_page.dart

**Comment by figbin** [comment]
Hello, I mean when the BetterPlayer is dispose, but the controller is keep, I have set autoDispose: false
if use controller.setupDataSource occur this error above!

step 1:
playerController.setupDataSource()
Container(child: showPlayer ? BetterPlayer(controller: playerController) : null)

step 2:
some button press call:
setState((){
showPlayer = false;
})

step 3:
playerController.setupDataSource()

you will found error above

**Comment by jhomlala** [comment]
Should be fixed in 0.0.61.
